omniNotes/src/androidTest/java/it/feio/android/omninotes/utils/EspressoUtils.java

```diff
@@ -90,54 +120,73 @@ static void writeSelectedUiObjectWithText(String textToBeFound, String textToBeR
         }
     }
 
-    public static String getTextFromSelectedUiObject(int index) throws UiObjectNotFoundException {
-
-            UiObject selectedDescription = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation()).findObject(new UiSelector().index(index));
-
-        return selectedDescription.getText();
-    }
-
+    /**
+     * this method checks if a text is completely displayed in a view
+     *
+     * @param p is the id used for searching the text
+     * @param s is the text searched
+     * @return true or false - depends if the text is or is not completely displayed
+     */
     public static boolean checkIfTextIsDisplayed(int p, String s) {
-        try{
+        try {
             onView(allOf(withId(p), withText(s))).check(matches(isCompletelyDisplayed()));
-        }catch (Exception e){
+        } catch (Exception e) {
             return false;
         }
         return true;
     }
 
+    /**
+     * this method is used for going back to the list of notes, after completing a note
+     */
     public static void clickBack() {
         onView(isRoot()).perform(waitForViewWithContentDescriptionIsDisplayed("drawer open", 2000));
         clickOnButton(backArrowButton);
     }
 
+    /**
+     * this method write text in a selected ViewInteraction
+     *
+     * @param stringToBeReplaced is the ViewInteraction searched in which the text will be written
+     * @param stringReplacer     is the text written
+     */
     static void writeText(ViewInteraction stringToBeReplaced, String stringReplacer) {
         stringToBeReplaced.perform(replaceText(stringReplacer));
     }
 
-    public static void deleteAllNotes(){
-
+    /**
+     * this method opens each note and clicks on Trash button until the list is empty
+     */
+    public static void deleteAllNotes() {
         boolean empty = false;
-        while(!empty) {
+        while (!empty) {
             try {
                 openNthNote(0);
                 clickOnTrash();
             } catch (Exception e) {
                 empty = true;
             }
         }
-
     }
 
+    /**
+     * this method adds the password when the pop-up with the password is displayed
+     */
     static void addPasswordWhenRequested() {
-        try{
-            writeText(onView(withId(R.id.password_request)),"parola");
+        try {
+            writeText(onView(withId(R.id.password_request)), "parola");
             clickOnButton(onView(withId(R.id.md_buttonDefaultPositive)));
-        }catch (Exception e){
+        } catch (Exception e) {
             Log.e("add password", "Exception when trying to add password");
         }
     }
 
+    /**
+     * this ViewAction waits a number of milliseconds for a View with a selected content description to be displayed
+     *
+     * @param contentDescription is the param used for the selected content description
+     * @param millis             is the number of milliseconds
+     */
```

Comment:
is this taken from somewhere?

Reply:
yes, I added the link from where I found it :D